DirectProgramming/C++SYCL_FPGA/ReferenceDesigns/matmul/README.md

```diff
@@ -0,0 +1,271 @@
+      ```
+      make report
+      ```
+      The report resides at `matmul_report/reports/report.html`.
+
+   3. Compile for FPGA hardware (longer compile time, targets FPGA device).
+      ```
+      make fpga
+      ```
+
+   (Optional) The hardware compiles listed above can take several hours to complete; alternatively, you can download FPGA precompiled binaries (compatible with Linux* Ubuntu* 18.04) from [https://iotdk.intel.com/fpga-precompiled-binaries/latest/matmul.fpga.tar.gz](https://iotdk.intel.com/fpga-precompiled-binaries/latest/matmul.fpga.tar.gz).
+
+### On Windows*
+
+>**Note**: The Intel® PAC with Intel Arria® 10 GX FPGA and Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX) do not yet support Windows*. Compiling to FPGA hardware on Windows* requires a third-party or custom Board Support Package (BSP) with Windows* support.
+
+1. Change to the sample directory.
+2. Configure the build system for **Intel® PAC with Intel Arria® 10 GX FPGA**, which is the default.
+   ```
+   mkdir build
+   cd build
+   cmake -G "NMake Makefiles" ..
+   ```
+   To compile for the **Intel® FPGA PAC D5005 (with Intel Stratix® 10 SX)**, enter the following:
+   ```
+   cmake -G "NMake Makefiles" .. -DFPGA_BOARD=intel_s10sx_pac:pac_s10
+   ```
+
+3. Compile the design. (The provided targets match the recommended development flow.)
+
+   1. Compile for emulation (fast compile time, targets emulated FPGA device).
+      ```
+      nmake fpga_emu
+      ```
+   2. Generate HTML performance report.
+      ```
+      nmake report
+      ```
+      The report resides at `matmul_report.a.prj/reports/report.html`.
+
+   3. Compile for FPGA hardware (longer compile time, targets FPGA device).
+      ```
+      nmake fpga
+      ```
+>**Note**: If you encounter any issues with long paths when compiling under Windows*, you may have to create your ‘build’ directory in a shorter path, for example `C:\samples\build`. You can then run cmake from that directory, and provide cmake with the full path to your sample directory.
+
+## Run the `Matrix Multiply` Design
+
+### Configurable Parameters
+
+| Argument  | Description
+|:---       |:---
+| `<num>`   | (Optional) Specifies the number of times to repeat the decomposition of a set of 8 matrices (only 1 matrix when running simulation). Its default value is **16** for the emulation flow, **1** for the simulation flow and **819200** for the FPGA flow.
+
+You can perform the multiplication of the set of matrices repeatedly. This step performs the following:
+- Generates the set of random matrices.
+- Computes the product of the set of matrices.
+- Repeats the multiplication multiple times (specified as a command line argument) to evaluate performance.
+
+### On Linux
+
+#### Run on FPGA Emulator
+
+1. Increase the amount of memory that the emulator runtime is permitted to allocate by exporting the `CL_CONFIG_CPU_FORCE_PRIVATE_MEM_SIZE` environment variable before running the executable.
```

Comment:
I'm not sure this is needed here, please remove

Reply:
removed